### ISSUE TITLE
Unable to signup/login with Chrome on ios

### ISSUE THREAD

**Issue by jayzeng** [sahat/satellizer#467]
All of our users who use chrome on ios (iphone, ipad etc) are not able to login or signup, I tested on the sample app and it works just fine (which uses v0.9.4)

I did a quick diff towards these two versions, looks like most changes are about localstorage and popup window. @sahat do you have any clue what changes may introduce this bug?

**Comment by sahat** [comment]
@jayzeng which version of Google Chrome on iOS are you using? I will try to reproduce it on my phone first.

**Comment by jayzeng** [comment]
@sahat I don't know see any way to check the version number, but it is the latest version that you can download from the app store, my colleague also says safari suffers from the same bug (but I am not able to reproduce it).

**Comment by montgomery1944** [comment]
The difference between 0.9.4 and the latest version that causes this issue is:
```
popupWindow = window.open(url, '_blank', optionsString);
```
was changed to:
```
popup.popupWindow = window.open(url, windowName, stringifiedOptions);
```
The problem is, that on all versions of Chrome on iOS to get a reference to opened window you can't assign any name different from "_blank" to window.

I was about to report this issue, but I just decided to disable functionality that used this library on iOS Chrome.

**Comment by sahat** [comment]
Hmm, so what's the best way to handle this problem? I have changed to `windowName` to avoid spawning multiple popups of the same provider if you try clicking on **Sign in with X** multiple times I could do something like `if (config.platform === 'mobile') { // use "blank" })` but wanted to double check if there is a better way to handle both mobile and desktop browsers at the same time.

**Comment by jayzeng** [comment]
@sahat I think what you proposed sounds reasonable to me, as I already have something like 

```
if(deviceDetector.isMobile) {
   $authProvider.platform = 'mobile';
}
```

**Comment by jayzeng** [comment]
popup.popupWindow = window.open(url, '_blank', stringifiedOptions);

**Comment by Pietro-Vis** [comment]
Yes, really strange that mobile signup works better usign platform browser actually. If i switch to mobile it doesn't works on multiple browser/os.
@sahat To address the problem of spawning multiple popups i modified the code of satellizer to set window.popupClosed = true every time that the popup is closed.
This way in our frontend we can poll window.popupClosed with a simple $interval and disable click on login buttons while popupClosed is still false.
Do you like the idea?

**Comment by sahat** [comment]
@Pietro-Vis @jayzeng would it be more clear if I renamed `platform: "mobile"` to `cordova: true`?

Right now "browser" refers to web browser flow, regardless if it is mobile browser or desktop browser. Whereas "mobile" refers to Ionic/Cordova flow.

**Comment by sahat** [comment]
> To address the problem of spawning multiple popups i modified the code of satellizer to set window.popupClosed = true every time that the popup is closed.
This way in our frontend we can poll window.popupClosed with a simple $interval and disable click on login buttons while popupClosed is still false.

@Pietro-Vis Wouldn't that prevent you to open any popup until the current one is closed? Or a better question, does it make sense to have both let's Facebook and Google popups open or should there only be one open popup at any given time?

**Comment by jayzeng** [comment]
Renaming platform: "mobile" to cordova: true will avoid all the confusion, I was under the impression if it does some mobile device (e.g: iphone, andriod etc) specific magics, turns out it is for cordova.

**Comment by Pietro-Vis** [comment]
@sahat 
I think platform mobile is confusing, i thought until now it was for browsing trough mobile web browser, not cordova. You could just replace it with platform = cordova.
Talking about popups, In our case there is no need at all to have multiple popups of different social networks opened together. After all the user just have to login/signup so he's going to open one popup at time. To me, to be able to open more popups at the same time would be just confusing for the user.

**Comment by sahat** [comment]
@Pietro-Vis could you please send a PR or post the code here for multiple popup fix? I would like to take a look at it first, and if it looks good I'll merge it.

**Comment by sahat** [comment]
@jayzeng I will change it to `cordova: true/false` in the next release. I don't imagine there will be more than two platforms any time soon, so having `platform: "some string"` is a bit unnecessary and more verbose.

**Comment by Pietro-Vis** [comment]
@sahat It's a bit difficult to make a pull request since we're actually using a customized fork of satellizer, but you can give a look to https://github.com/sahat/satellizer/commit/2b07065c2d75221340d7fd9f94e696b974b58af5 just to have an idea of what i mean.

**Comment by sahat** [comment]
Resolved in [0.12](https://github.com/sahat/satellizer/releases/tag/0.12.0) release.
